param/parameterized.py

```diff
@@ -2700,15 +2693,17 @@ def _register_watcher(self_, action, watcher, what='value'):
                 raise ValueError("{} parameter was not found in list of "
                                  "parameters of class {}".format(parameter_name, self_.cls.__name__))
 
-            if self_.self is not None and what == "value":
-                watchers = self_.self._param__private.watchers
+            inst = self_.self
```

Comment:
Any reason to create `inst` now?

Reply:
Not really can revert all of this I guess. Still reads a little clearer but not much.